src/renderer/views/swap/SwapView.tsx

```diff
@@ -129,41 +125,39 @@ export const SwapView: React.FC<Props> = (_): JSX.Element => {
   const oSource = useMemo(() => O.fromNullable(assetFromString(source.toUpperCase())), [source])
   const oTarget = useMemo(() => O.fromNullable(assetFromString(target.toUpperCase())), [target])
 
-  const sourceChainFee$: ChainFeeLD = useMemo(
+  const swapFeesLDS: O.Option<SwapFeesLDS> = useMemo(
     () =>
       FP.pipe(
-        oSource,
-        O.map((asset) =>
-          FP.pipe(
-            feesByAssetChain$(asset),
-            liveData.map((fees) => ({ chainAsset: getChainAsset(asset.chain), amount: fees.fastest }))
-          )
-        ),
-        O.getOrElse((): ChainFeeLD => Rx.of(RD.initial))
+        sequenceTOption(oSource, oTarget),
+        O.map(([sourceAsset, targetAsset]) => swapFees$(sourceAsset, targetAsset))
       ),
-    [oSource, feesByAssetChain$]
+    [oSource, oTarget, swapFees$]
+  )
+  const sourceFee: SwapFeeRD = useObservableState(
+    FP.pipe(
+      swapFeesLDS,
+      O.map((fees) => fees.source),
+      O.getOrElse((): SwapFeeLD => Rx.EMPTY)
+    ),
+    RD.initial
   )
 
-  const targetChainFee$: ChainFeeLD = useMemo(
-    () =>
-      FP.pipe(
-        oTarget,
-        O.map((asset) =>
-          FP.pipe(
-            feesByAssetChain$(asset),
-            liveData.map((fees) => ({
-              chainAsset: getChainAsset(asset.chain),
-              amount: baseAmount(fees.fastest.amount().times(3))
-            }))
-          )
-        ),
-        O.getOrElse((): ChainFeeLD => Rx.of(RD.initial))
-      ),
-    [oTarget, feesByAssetChain$]
+  const targetFee: SwapFeeRD = useObservableState(
+    FP.pipe(
+      swapFeesLDS,
+      O.map((fees) => fees.target),
+      O.getOrElse((): SwapFeeLD => Rx.EMPTY)
+    ),
+    RD.initial
   )
 
-  const targetChainFee = useObservableState(sourceChainFee$)
-  const sourceChainFee = useObservableState(targetChainFee$)
+  const swapFeesRDS: SwapFeesRDS = useMemo(
+    () => ({
+      source: sourceFee,
+      target: targetFee
+    }),
+    [sourceFee, targetFee]
+  )
```

Comment:
^ With suggested changes before, no need to create `targetFee: SwapFeeRD` / `sourceFee: SwapFeeRD` in this view anymore.